### PR TITLE
OOM protected docker processes (-900)

### PR DESCRIPTION
**Pull request by dchen1107**
Also changed ApplyOomScore util function to return error. 
#7376, #7795

### CODE REVIEW

File: pkg/kubelet/container_manager_linux.go [hunk 1]
```diff
@@ -39,7 +39,8 @@ type containerManagerImpl struct {
 	dockerContainerName string
 
 	// The manager of the resource-only container Docker should be in.
-	manager fs.Manager
+	manager     fs.Manager
+	oomScoreAdj int
```
Comment (vmarmol) [inline pkg/kubelet/container_manager_linux.go]: nit: dockerOomScoreAdj since we may do it for others

**Comment by vmarmol** [comment]
Just a nit, otherwise LGTM

**Comment by dchen1107** [comment]
Addressed your comments. Thanks!

**Comment by vmarmol** [comment]
LGTM

**Comment by vmarmol** [comment]
thanks @dchen1107!

**Pull request merged**